wagtail/admin/templates/wagtailadmin/shared/header.html

```diff
@@ -8,19 +8,24 @@
     - `search_url` - if present, display a search box. This is a URL route name (taking no parameters) to be used as the action for that search box
     - `query_parameters` - a query string (without the '?') to be placed after the search URL
     - `icon` - name of an icon to place against the title
+    - `avatar` - if present, display an 'avatar' in place of icon. This is the URL to be used as the img src for avatar 
     - `merged` - if true, add the classname 'merged'
     - `action_url` - if present, display an 'action' button. This is the URL to be used as the link URL for the button
     - `action_text` - text for the 'action' button
     - `action_icon` - icon for the 'action' button, default is 'icon-plus'
 
 {% endcomment %}
-<header class="header {% if merged %}merged{% endif %} {% if search_form %}hasform{% endif %}">
+<header class="header {{ classname }} {{ merged|yesno:" merged ," }}{{ search_form|yesno:" hasform ," }}">
```

Comment:
```suggestion
<header class="header {{ classname }} {{ merged|yesno:" merged ," }}{{ search_form|yesno:" hasform ," }}">
```

thinking long term - it might be nice to have a generic template tag for classnames (similar to https://www.npmjs.com/package/classnames)

```
<header class="{% classnames "header" classname merged|yesno:" merged," search_form|yesno:"hasform," %}">
```

Whatever you give it as args, it will try to parse and spit out a nice classname string without extra spaces.

Give this a go.


----

https://docs.djangoproject.com/en/4.0/howto/custom-template-tags/#django.template.Library.simple_tag

use simple tags, something like...

```python
register = template.Library()

@register.simple_tag
def classnames(*classes):
    #classes # will be an array
    return ' '.join([classname for classname in classes if classname])
```

It will need a test case or two but could come in handy.

Reply:
Just curious why we're going to so much trouble here to strip the extra spaces? (Ease of writing test assertions, perhaps?)

Reply:
Yeah. It's been a pain in a bunch of previous unit tests.

Reply:
I'll try this approach soon. Other than that it's ready for review.

Reply:
Thanks, not critical, only go down this path if we get some benefits from it.

Reply:
Tried it but It is not working. Getting the same output `class="header  merged"` i.e with double spaces.

Reply:
Be careful what you are passing into the template tag. 

```
<header class="header {{ classname }} {{ merged|yesno:" merged ," }}{{ search_form|yesno:" hasform ," }}">

```

This will add a space before the merged string as is. 

But...
```
<header class="header {{ classname }} {{ merged|yesno:"merge ," }}{{ search_form|yesno:" hasform ," }}">
```

Will not. 

Have a read about how yesno works - it will output whatever is inside the supplied string after yesno: (before the comma) if true. 

https://docs.djangoproject.com/en/4.0/ref/templates/builtins/#yesno

Reply:
Comment above fixed up - code snippets pasted wrong initially